src/date_type.h

```diff
@@ -92,8 +97,8 @@ static const TimerGameCalendar::Year DEF_END_YEAR = ORIGINAL_END_YEAR - 1;
  */
 static const TimerGameCalendar::Year MAX_YEAR  = 5000000;
 
-/** The number of days till the last day */
-#define MAX_DAY (DAYS_TILL(MAX_YEAR + 1) - 1)
+/** The number of days till the last day, starting at day 0. */
+static const TimerGameCalendar::Day MAX_DAY = DaysTill(MAX_YEAR + 1) - 1;
```

Comment:
`Day` is `uint8`, `MAX_YEAR` is way bigger, and so is number of days in a year. I suggest renaming to `MAX_DATE`.

Reply:
Ah yes, meant this to be Date but confused the two. Problems with submitting PRs after a long day of chasing the bottom of rabbitholes. 😄